pycbc/inference/sampler_emcee.py

```diff
@@ -0,0 +1,996 @@
+        return fp.attrs['acl']
+
+    @staticmethod
+    def read_acls(fp):
+        """Reads the acls of all the walker chains saved in the given file.
+
+        Parameters
+        ----------
+        fp : InferenceFile
+            An open file handler to read the acls from.
+
+        Returns
+        -------
+        WaveformArray
+            An ntemps x nwalkers `WaveformArray` containing the acls for
+            every temp and walker, with the variable arguments as fields.
+        """
+        group = fp.samples_group + '/{param}/temp{tk}/walker{wi}'
+        widx = numpy.arange(fp.nwalkers)
+        tidx = numpy.arange(fp.ntemps)
+        arrays = {}
+        for param in fp.variable_args:
+            arrays[param] = numpy.array([
+                [fp[group.format(param=param, tk=tk, wi=wi)].attrs['acl']
+                    for wi in widx]
+                 for tk in tidx])
+        return WaveformArray.from_kwargs(**arrays)
+
+    @classmethod
+    def calculate_logevidence(cls, fp, thin_start=None, thin_end=None,
+            thin_interval=None):
+        """Calculates the log evidence from the given file using emcee's
+        thermodynamic integration.
+
+        Parameters
+        ----------
+        fp : InferenceFile
+            An open file handler to read the stats from.
+        thin_start : int
+            Index of the sample to begin returning stats. Default is to read
+            stats after burn in. To start from the beginning set thin_start
+            to 0.
+        thin_interval : int
+            Interval to accept every i-th sample. Default is to use the
+            `fp.acl`. If `fp.acl` is not set, then use all stats
+            (set thin_interval to 1).
+        thin_end : int
+            Index of the last sample to read. If not given then
+            `fp.niterations` is used.
+        
+        Returns
+        -------
+        lnZ : float
+            The estimate of log of the evidence.
+        dlnZ : float
+            The error on the estimate.
+        """
+        try:
+            import emcee
+        except ImportError:
+            raise ImportError("emcee is not installed.")
+
+        logstats = cls.read_likelihood_stats(fp, thin_start=thin_start,
+            thin_end=thin_end, thin_interval=thin_interval,
+            temps='all', flatten=False)
+        # get the likelihoods
+        logls = logstats['loglr'] + fp.lognl
+        # we need the betas that were used
+        betas = fp.attrs['betas']
+        # annoyingly, theromdynaimc integration in PTSampler is an instance
+        # method, so we'll implement a dummy one
+        ntemps = fp.ntemps
+        nwalkers = fp.nwalkers
+        ndim = len(fp.variable_args)
+        dummy_sampler = emcee.PTSampler(ntemps, nwalkers, ndim, None,
+                                        None, betas=betas)
+        return dummy_sampler.thermodynamic_integration_log_evidence(
+                    logls=logls, fburnin=0.)
+
```

Comment:
__[flake8]__
* [W391] blank line at end of file

<!-- Comment by [SideCI](https://sideci.com) -->